nextdns/schema_nextdns_profile.go

```diff
@@ -16,5 +16,67 @@ func resourceNextDNSProfileSchema() map[string]*schema.Schema {
 			Type:        schema.TypeString,
 			Required:    true,
 		},
+		"endpoint": {
+			Description: "Endpoints.",
+			Type:        schema.TypeList,
+			Computed:    true,
+			Elem: &schema.Resource{
```

Comment:
I don't understand this being nested as a list, won't it always only have a single element?

As in user will always need to do `next_dns.profile.foo.endpoint[0].doh` instead of `next_dns.profile.foo.endpoint.doh`, and there could never be an `endpoint[1]`?

Reply:
I agree it makes it easier and more natural to access resources like `nextdns_profile.foo.endpoint_dot` than `nextdns_profile.foo.endpoint[0].dot`, let me sleep on it a bit more.

Reply:
Ah I see. Fwiw it is possible in the new 'framework' (SDK v3 by a different name basically), but that's quite a big change just to support this 😅.

Another option then if you didn't want to pollute the top level could be a separate data source, used like:
```tf
data "nextdns_endpoint" "foo" {
  profile = nextdns_profile.foo.profile_id
}

locals {
  doh = data.nextdns_endpoint.foo.doh
}
```

Reply:
Good to know, but I think it's best with data sources for each type of resource, since it isn't something we can enforce but only consume, so having a data source for `nextdns_endpoint` and `nextdns_linkedip` could make more sense here. I will try it later today.